PerfettoCds/Pipeline/Events/PerfettoSqlEvent.cs

```diff
@@ -0,0 +1,490 @@
+﻿using Microsoft.Performance.SDK;
+using Microsoft.Performance.SDK.Extensibility;
+using System;
+using Perfetto.Protos;
+
+namespace PerfettoCds.Pipeline.Events
+{
+    /// <summary>
+    /// Keep track of where we are in the varint, varfloat, and varstring arrays in a Perfetto batch result
+    /// </summary>
+    public class CellCounters
+    {
+        public int IntCounter { get; set; }
+        public int StringCounter { get; set; }
+        public int FloatCounter { get; set; }
+    }
+
+    /// <summary>
+    /// Base class for a row in a Perfetto SQL table
+    /// </summary>
+    public abstract class PerfettoSqlEvent : IKeyedDataType<String>
+    {
+        public readonly string Key;
+
+        public PerfettoSqlEvent(string key)
+        {
+            this.Key = key;
+        }
+
+        public int CompareTo(string other)
+        {
+            return this.Key.CompareTo(other);
+        }
+
+        public string GetKey()
+        {
+            return Key;
+        }
+
+        /// <summary>
+        /// This method is responsible for processing an individual cell in a SQL query, which is stored in the QueryResult
```

Comment:
Comment / Link out to any Perfetto docs that mention the same or reference info for this